third_party/bigframes_vendored/google_cloud_bigquery/_pyarrow_helpers.py

```diff
@@ -0,0 +1,115 @@
+# Original: https://github.com/googleapis/python-bigquery/blob/main/google/cloud/bigquery/_pyarrow_helpers.py
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Shared helper functions for connecting BigQuery and pyarrow."""
+
+from typing import Any
+
+import pyarrow  # type: ignore
+
+
+def pyarrow_datetime():
+    return pyarrow.timestamp("us", tz=None)
+
+
+def pyarrow_numeric():
+    return pyarrow.decimal128(38, 9)
+
+
+def pyarrow_bignumeric():
+    # 77th digit is partial.
+    # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types
+    return pyarrow.decimal256(76, 38)
+
+
+def pyarrow_time():
+    return pyarrow.time64("us")
+
+
+def pyarrow_timestamp():
+    return pyarrow.timestamp("us", tz="UTC")
+
+
+_BQ_TO_ARROW_SCALARS = {}
+_ARROW_SCALAR_IDS_TO_BQ = {}
+
+# This dictionary is duplicated in bigquery_storage/test/unite/test_reader.py
```

Comment:
This file doesn't seem to be added yet

Reply:
This is copied directly from upstream https://github.com/googleapis/python-bigquery repo.

It's referring to this file: https://github.com/googleapis/python-bigquery-storage/blob/main/tests/unit/test_reader_v1_arrow.py#L44

Note: I have proposed go/pandas-gbq-and-bigframes-redundancy so that we won't have to keep this copy around. Will chat with the BigQuery SDK folks more about it on Thursday.

Reply:
Got it, thanks